### ISSUE TITLE
Don't error on "targetDependencies" "self" in nx.json

### ISSUE THREAD

**Issue by flippidippi** [nrwl/nx#9618]
- [] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description

Currently, we have libs that lint and test depend on the codegen target that lives in those projects.
We have to make sure to add "dependsOn" in each of these libs to make sure "codegen" is run before lint and test.
We would like to instead be able to do this in `nx.json`, but currently nx throws an error if "targetDependencies" is "self" and the project doesn't contain that target.

## Motivation

Easier maintainability for depends on targets for "self".

## Suggested Implementation

Don't throw an error if the "targetDependencies" "self" is not a target in the project.

```
  ...
  "targetDependencies": {
    "build": [
      {
        "target": "build",
        "projects": "dependencies"
      },
      {
        "target": "codegen",
        "projects": "dependencies"
      },
      {
        "target": "codegen",
        "projects": "self" // < THROWS IF NOT TARGET IN CURRENT PROJECT
      }
    ],
    ...
```

## Alternate Implementations

A new target like "self-optional" or something to indicate to not throw if the target is not there.

**Comment by flippidippi** [comment]
We are on 13.2.2

**Comment by AgentEnder** [comment]
Ah, gotcha. Yeah this should be fixed in 13.9.X (perhaps one of the later 13.8.X's, not quite sure on exact version)
